dwave_networkx/generators/chimera.py

```diff
@@ -356,147 +357,119 @@ def __init__(self, m, n=None, t=4):
+               'dwave-networkx 0.9.0, please use '
+               'chimera_coordinates.chimera_to_linear instead')
+        warnings.warn(msg, DeprecationWarning)
+        return self.chimera_to_linear(q)
+
+    def chimera_to_linear(self, q):
+        """Convert a 4-term Chimera coordinate to a linear index."""
         i, j, u, k = q
         m, n, t = self.args
         return ((n*i + j)*2 + u)*t + k
 
     def tuple(self, r):
-        """
-        Converts the linear_index `q` into an chimera_index
-
-        Parameters
-        ----------
-        r : int
-            The linear_index node label    
-
-        Returns
-        -------
-        q : tuple
-            The chimera_index node label corresponding to r
-        """
-
+        """Deprecated alias for `linear_to_chimera`."""
+        msg = ('chimera_coordinates.tuple is deprecated and will be removed in '
+               'dwave-networkx 0.9.0, please use '
+               'chimera_coordinates.linear_to_chimera instead')
+        warnings.warn(msg, DeprecationWarning)
+        return self.linear_to_chimera(r)
+
+    def linear_to_chimera(self, r):
+        """Convert a linear index to a 4-term Chimera coordinate."""
         m, n, t = self.args
         r, k = divmod(r, t)
         r, u = divmod(r, 2)
         i, j = divmod(r, n)
         return i, j, u, k
 
     def ints(self, qlist):
+        """Deprecated alias for `iter_chimera_to_linear`."""
+        msg = ('chimera_coordinates.ints is deprecated and will be removed in '
+               'dwave-networkx 0.9.0, please use '
+               'chimera_coordinates.iter_chimera_to_linear instead')
+        warnings.warn(msg, DeprecationWarning)
+        return self.iter_chimera_to_linear(qlist)
+
+    def iter_chimera_to_linear(self, qlist):
+        """Return an iterator converting 4-term Chimera coordinates to linear
+        indices.
         """
-        Converts a sequence of chimera_index node labels into
-        linear_index node labels, preserving order
-
-        Parameters
-        ----------
-        qlist : sequence of ints
-            The chimera_index node labels
-
-        Returns
-        -------
-        rlist : iterable of tuples
-            The linear_lindex node lables corresponding to qlist
-        """
-
         m, n, t = self.args
-        return (((n*i + j)*2 + u)*t + k for (i, j, u, k) in qlist)
+        for (i, j, u, k) in qlist:
+            yield ((n*i + j)*2 + u)*t + k
 
     def tuples(self, rlist):
+        """Deprecated alias for `iter_linear_to_chimera`."""
+        msg = ('chimera_coordinates.tuples is deprecated and will be removed in '
+               'dwave-networkx 0.9.0, please use '
+               'chimera_coordinates.iter_linear_to_chimera instead')
+        warnings.warn(msg, DeprecationWarning)
+        return self.iter_linear_to_chimera(rlist)
+
+    def iter_linear_to_chimera(self, rlist):
+        """Return an iterator converting linear indices to 4-term Chimera
+        coordinates.
```

Comment:
I think this needs to specify that the input is a list, something like, "converting a list of linear"